inference-engine/src/mkldnn_plugin/nodes/mkldnn_strided_slice_node.h

```diff
@@ -24,58 +25,73 @@ class MKLDNNStridedSliceNode : public MKLDNNNode {
-    std::vector<int> endMask;
-    std::vector<int> ellipsisMask;
-    std::vector<int> newAxisMask;
-    std::vector<int> shrinkAxisMask;
-
-    InferenceEngine::SizeVector beginDims;
-    InferenceEngine::SizeVector endDims;
-    InferenceEngine::SizeVector strideDims;
-
-    struct {
-        MKLDNNMemoryPtr srcMemPtr = nullptr;
-        MKLDNNMemoryPtr dstMemPtr = nullptr;
-        InferenceEngine::SizeVector srcDims;
-        InferenceEngine::SizeVector dstDims;
-        InferenceEngine::SizeVector srcStrides;
-        InferenceEngine::SizeVector dstStrides;
-        InferenceEngine::SizeVector srcIndices;
-        InferenceEngine::SizeVector dstIndices;
-        int ellipsisPos1 = -1;
-        int ellipsisPos2 = 0;
-        size_t nThreads = 0;
-        size_t nDimsForWork = 0;
-        size_t workAmount = 0;
-        size_t lastDstDim = 0;
-        size_t dataSize = 0;
-        size_t srcShift = 0;
-        bool isOptimized = false;
+    void addHiddenDims(const size_t nSrcDims, int ellipsisPos1);
+    void orderParametersByLayouts(const MKLDNNMemoryPtr& srcMemPtr);
+
+    struct StridedSliceAttributes {
+        std::vector<int> begin;
+        std::vector<int> end;
+        std::vector<int> stride;
+
+        std::vector<int> beginMask;
+        std::vector<int> endMask;
+        std::vector<int> ellipsisMask;
+        std::vector<int> newAxisMask;
+        std::vector<int> shrinkAxisMask;
+
+        InferenceEngine::SizeVector beginDims;
+        InferenceEngine::SizeVector endDims;
+        InferenceEngine::SizeVector strideDims;
+
         bool equalDims = false;
-        bool parametersAreConstant = true;
-    } params;
+        size_t dataSize = 1lu;
+    } attrs;
+
+    struct StridedSliceExecutor {
+        StridedSliceExecutor(const StridedSliceAttributes& attrs, const InferenceEngine::SizeVector& srcDims, const InferenceEngine::SizeVector& dstDims);
+        void exec(const uint8_t* srcData, uint8_t* dstData);
+        ~StridedSliceExecutor() = default;
+
+    private:
+        void dimsNormalization(InferenceEngine::SizeVector& newSrcDims, InferenceEngine::SizeVector& newDstDims);
+        void dimsGluing(const size_t realNDims, const InferenceEngine::SizeVector& newSrcDims, const InferenceEngine::SizeVector& newDstDims);
+        void indicesCalculation();
+        void indicesCalculationForOptimized();
+
+        struct {
+            StridedSliceAttributes attrs;
+
+            InferenceEngine::SizeVector srcDims;
+            InferenceEngine::SizeVector dstDims;
+            InferenceEngine::SizeVector srcStrides;
+            InferenceEngine::SizeVector dstStrides;
+            InferenceEngine::SizeVector srcIndices;
+            InferenceEngine::SizeVector dstIndices;
+
+            size_t nThreads = 0lu;
+            size_t nDimsForWork = 0lu;
+            size_t workAmount = 0lu;
+            size_t lastDstDim = 0lu;
+            size_t srcShift = 0lu;
+
+            bool isOptimized = false;
+        } params;
```

Comment:
As far as I understand only part of this params data is used in the exec call. Do we really need to store all these data, especially attrs inside the executor objects?

Reply:
No, we don't. I left only necessary attributes for execution and created structure with useful parameters (strides, dims, etc) to comfortable work inside executor.
Good catch! Thank you!